From f283252018b4b57df444f8d68e8e8c595d358495 Mon Sep 17 00:00:00 2001
From: Liang Chen <cl@rock-chips.com>
Date: Wed, 7 Feb 2018 15:06:22 +0800
Subject: [PATCH] clk: rockchip: px30: make clk_gpu_divnp5 critical

If disable clk_gpu_divnp5, we will get error below:

[    3.699114] rockchip-pm-domain ff000000.power-management:power-controller: failed to get ack on domain
 'pd_gpu', val=0x43c443c4
[    3.700579] W : [File] : drivers/gpu/arm/bifrost/platform/rk/mali_kbase_config_rk.c; [Line] : 134; [Fu
nc] : kbase_platform_rk_init(); power-off-delay-ms not available.
[    3.700880] Unhandled fault: synchronous external abort (0x96000010) at 0xffffff8009620000
[    3.700897] Internal error: : 96000010 [#1] PREEMPT SMP
[    3.703090] Modules linked in: bifrost_kbase(+)
[    3.703524] CPU: 0 PID: 1 Comm: init Not tainted 4.4.112 #578
[    3.704034] Hardware name: Rockchip rk3326 evb board (DT)
[    3.704515] task: ffffffc00a308000 task.stack: ffffffc00a310000
[    3.705362] PC is at kbase_reg_read+0xd0/0x1c8 [bifrost_kbase]
[    3.706173] LR is at kbase_backend_gpuprops_get+0x24/0x20c [bifrost_kbase]

Change-Id: I481cdaa60b5174ae6763e11fb42f79ed9208c120
Signed-off-by: Liang Chen <cl@rock-chips.com>
---
 drivers/clk/rockchip/clk-px30.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/clk/rockchip/clk-px30.c b/drivers/clk/rockchip/clk-px30.c
index 842050b59e26..a6d6492821be 100644
--- a/drivers/clk/rockchip/clk-px30.c
+++ b/drivers/clk/rockchip/clk-px30.c
@@ -1007,6 +1007,7 @@ static const char *const px30_critical_clocks[] __initconst = {
 	"aclk_peri_pre",
 	"hclk_peri_pre",
 	"aclk_gpu_niu",
+	"clk_gpu_divnp5",
 };
 
 static const char *const px30_pmucru_critical_clocks[] __initconst = {
-- 
2.35.3

